astropy/coordinates/representation.py

```diff
@@ -436,8 +439,11 @@ def __init__(cls, name, bases, dct):
         repr_name = cls.get_name()
 
         if repr_name in REPRESENTATION_CLASSES:
-            raise ValueError("Representation class {0} already defined"
-                             .format(repr_name))
+            msg = 'Representation {} already defined, using name "{}"'.format(
+                repr_name, cls.__qualname__
+            )
+            warnings.warn(msg, DuplicateRepresentationWarning)
```

Comment:
I think you still need to allow for the possibility that even the `__qualname__`'s are the same. So that would be something like:

```suggestion
            if cls.__qualname__ in REPRESENTATION_CLASSES:
                raise ValueError("Representation class {0} already defined".format(cls.__qualname__)
            warnings.warn(msg, DuplicateRepresentationWarning)
```

This is more of an edge-case than the previous one, but it's still possible when weird import things are happening and we may as well catch it just in case.